pytest_fixtures/subscription_fixtures.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+
+from robottelo.config import settings
+from robottelo.constants import RHEL_7_MAJOR_VERSION
+from robottelo.rhsso_utils import run_command
+
+
+def unsubscribe():
+    """unregisters a machine from cdn"""
+    run_command('subscription-manager unregister')
+    run_command('subscription-manager clean')
+
+
+@pytest.fixture(scope='session')
+def clean_rhsm():
+    """removes pre-existing candlepin certs and resets RHSM."""
+    # removing the katello-ca-consumer
+    run_command('rpm -qa | grep katello-ca-consumer | xargs -r rpm -e')
+
+    # resetting rhsm.conf to point to cdn.
+    run_command(
+        "sed -i -e 's/^hostname.*/hostname=subscription.rhsm.redhat.com/' " "/etc/rhsm/rhsm.conf"
+    )
+    run_command("sed -i -e 's|^prefix.*|prefix=/subscription|' /etc/rhsm/rhsm.conf")
+    run_command("sed -i -e 's|^baseurl.*|baseurl=https://cdn.redhat.com|' " "/etc/rhsm/rhsm.conf")
+    run_command(
+        "sed -i -e "
+        "'s/^repo_ca_cert.*/repo_ca_cert=%(ca_cert_dir)sredhat-uep.pem/' "
+        "/etc/rhsm/rhsm.conf"
+    )
+
+
+@pytest.fixture(scope='session')
+def subscribe_satellite(clean_rhsm):
+    """subscribe satellite to cdn"""
+    run_command(
+        'subscription-manager register --force --user={} --password={} {}'.format(
+            settings.subscription.rhn_username,
+            settings.subscription.rhn_password,
+            # set release to "7Server" currently with this scope
+            f'--release="{RHEL_7_MAJOR_VERSION}Server"',
```

Comment:
I think this needs to be more dynamic, as the version of RHEL that Satellite is running on will change.

There is a property that you can use to actually look this up at run-time, instead of hard coding it.

Include `default_sat` in the fixture parameters, and then reference `default_sat.os_version`.  This will give you a `packaging.Version` instance, which has properties for `major`, `minor`, etc.  `default_sat.os_version.major` will give you the major RHEL version that the target satellite is running on.

Reply:
@pondrejk why not write an issue to track this request if you do not have time to do it in this sprint?